### PR TITLE
Clarify the CI issue template

### PR DESCRIPTION
**Pull request by kytrinyx**
This template assumed a lot of knowledge about Exercism exercises and
test suites. For people who are implementing a new track without much
prior experience with Exercism, this was pretty confusing.

@jtigger @parkerl @IanWhitney Would you help verify this for clarity?

### CODE REVIEW

File: TRAVIS [hunk 1]
```diff
@@ -1,11 +1,41 @@
...
+
+This is the same experience that they would get if they were implementing each new test themselves.
+
+Most testing frameworks do not have the necessary configuration options to get this behavior directly, but they often do have a way of marking tests as _skipped_ or _pending_. The mechanism for this will vary from language to language and from test framework to test framework.
+
+Whatever the mechanism—functions, methods, annotations, directives, or some other approach—these are text added to the test file. The person solving the exercise will need to edit the test file in order to "activate" each subsequent test.
```
Comment (parkerl) [inline TRAVIS]: Not sure where it goes, but I imagine in some cases all you can do is comment out the entire test.

File: TRAVIS [hunk 2]
```diff
@@ -1,11 +1,41 @@
...
+
+Any tests that are marked as skipped will not be verified by the track test suite unless special care is taken.
+
+Additionally, in some programming languages, the name of the file containing the solution is hard-coded in the test suite, and the example solution is not named in the way that we expect people to name their files.
+
+We will need to temporarily (and programmatically) edit the exercise test suites to ensure that all of their tests are active. We may also need to rename the example solution file(s) in order for the exercise test suite to run against it.
```
Comment (parkerl) [inline TRAVIS]: Should there be links to example build files?

File: TRAVIS [hunk 3]
```diff
@@ -1,11 +1,41 @@
...
+
+**Avoiding accidental git check-ins**
+
+It's important that if we rewrite files in any way during a test run, that these changes do not accidentally get checked in to the git repository.
+
+Therefore, many language tracks write the track test suite in such a way that it _copies_ the exercise to a temporary location outside of the git repository before editing or rewriting the exercise files during a test run.
+
+**Working around long-running track test suites**
+
+Usually as people are developing the track, they're focused on a single exercise. If running the entire track test suite against all of the exercises takes a long time, it is often worth making it possible to verify just one exercise at a time.
```
Comment (parkerl) [inline TRAVIS]: Do you know of an example of this?

**Comment by parkerl** [comment]
I wonder if in this file or maybe the TRACK_README we should emphasize that the _tests_ are really the fundamental "product" that Exercism delivers. Utmost care should be taken when crafting a track's test suite. It might also be good to note in CONSISTENCY.

Additionally might it be good to note that actually TDDing an exercise from scratch is a good way to build tests that mimic that flow as closely as possible for a user? I sometimes worry that `cannonical_data` handcuffs tracks a bit. That is a topic for another place.

File: TRAVIS (hunk 3 above)
Comment (kytrinyx) [inline TRAVIS]: Yeah, the tests for the ruby track take a long time to run (about 16 seconds on my machine last time I ran them). I don't have the patience to wait for that if I'm just working on one exercise.

**Comment by kytrinyx** [comment]
> we should emphasize that the tests are really the fundamental "product" that Exercism delivers.

Yes, absolutely. I think it makes sense to put this in the track README, which will be seen by everyone, not in this issue, which will only be seen by the folks kicking off the track.

> [canonical data vs tdd] is a topic for another place.

Yeah, that's definitely worth discussing at some point.

File: TRAVIS (hunk 1 above)
Comment (kytrinyx) [inline TRAVIS]: We've not run into it yet, but yeah, that might be necessary at some point. I'll update this.

File: TRAVIS (hunk 2 above)
Comment (kytrinyx) [inline TRAVIS]: Yeah, there should be. I'll find something to link to.

**Comment by kytrinyx** [comment]
OK, I've updated this based on your feedback, @parkerl

File: TRAVIS [hunk 4]
```diff
@@ -1,11 +1,48 @@
-Implement Continuous Integration
+Implement continuous integration
 
-Implement a test suite that can run both locally and on Travis CI, that verifies that the example code passes the test suite for each exercise.
+Implement a track test suite that can run both locally and on Travis CI. The track test suite should verify that each exercise makes sense, by running the exercise tests against the example solution.
 
-Things to consider:
+**Definition of terms**
```
Comment (jtigger) [inline TRAVIS]: 👍 I really struggled in describing this in the Java track; love the terms.

**Comment by kytrinyx** [comment]
@jtigger Thanks! I really struggled trying to put words to it, so I don't doubt that we could improve it.

I'm going to merge this in as is for now, and open a new issue with your feedback so that you/we/someone can take a stab at improving the flow later.

**Comment by jtigger** [comment]
Yeah, it pains me to give "feedback" _without_ an example, so I'm itchin' to help on that front.  You're nailing the _hard_ part with this PR: naming the essential and omitting the unessential.

**Pull request merged**